blend-renderer/src/main/java/com/esri/arcgisruntime/sample/blendrenderer/MainActivity.java

```diff
@@ -0,0 +1,195 @@
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+    //set default values for blend parameters
+    mAltitude = 45;
+    mAzimuth = 315;
+    mZFactor = 0.000016;
+    mSlopeType = SlopeType.NONE;
+    mColorRampType = ColorRamp.PresetType.NONE;
+    mPixelSizeFactor = 1;
+    mPixelSizePower = 1;
+    mOutputBitDepth = 8;
+    // retrieve the MapView from layout
+    mMapView = (MapView) findViewById(R.id.mapView);
+    mFragmentManager = getSupportFragmentManager();
+    // define permission to request
+    String[] reqPermission = new String[]{ Manifest.permission.WRITE_EXTERNAL_STORAGE};
+    int requestCode = 2;
+    // For API level 23+ request permission at runtime
+    if (ContextCompat.checkSelfPermission(MainActivity.this,
+        reqPermission[0]) == PackageManager.PERMISSION_GRANTED) {
+      blendRenderer();
+    } else {
+      // request permission
+      ActivityCompat.requestPermissions(MainActivity.this, reqPermission, requestCode);
+    }
+  }
+
+  /**
+   * Using values stored in strings.xml, builds path to rasters.
+   * @return the path to raster file
+   */
+  private String buildRasterPath(String filename) {
+    // get sdcard resource name
+    File extStorDir = Environment.getExternalStorageDirectory();
+    // get the directory
+    String extSDCardDirName =
+        this.getResources().getString(R.string.data_sdcard_offline_dir);
+    // create the full path to the raster file
+    return extStorDir.getAbsolutePath()
+        + File.separator
+        + extSDCardDirName
+        + File.separator
+        + filename
+        + ".tif";
+  }
+
+  /**
+   * Creates new imagery and elevation rasters based on a given path, creates an ArcGISMap, sets it to a MapView and
+   * calls updateRenderer().
+   */
+  private void blendRenderer() {
+    // create rasters
+    mImageryRaster = new Raster(
+        new File(buildRasterPath(this.getString(R.string.imagery_raster_name))).getAbsolutePath());
+    mElevationRaster = new Raster(
+        new File(buildRasterPath(this.getString(R.string.elevation_raster_name))).getAbsolutePath());
+    // create a map
+    ArcGISMap map = new ArcGISMap();
+    // add the map to a map view
+    mMapView.setMap(map);
+    updateRenderer();
+  }
+
+  /**
+   * Creates ColorRamp and BlendRenderer according to the chosen property values.
+   */
+  private void updateRenderer() {
+    ColorRamp colorRamp = mColorRampType != ColorRamp.PresetType.NONE ? new ColorRamp(mColorRampType, 800) : null;
+    // if color ramp is not NONE, color the hillshade elevation raster instead of using satellite imagery raster color
+    mRasterLayer = colorRamp != null ? new RasterLayer(mElevationRaster) : new RasterLayer(mImageryRaster);
+    mMapView.getMap().setBasemap(new Basemap(mRasterLayer));
+      // create blend renderer
+      BlendRenderer blendRenderer = new BlendRenderer(
+          mElevationRaster,
+          Arrays.asList(9.0),
+          Arrays.asList(255.0),
```

Comment:
Prefer `Collections.singletonList(255.0)`, immutable over `Arrays.asList()` to create a single item list.